contracts/upgradable/BaseProxy.sol

```diff
@@ -5,26 +5,49 @@ pragma solidity ^0.8.0;
 import { IProxy } from '../interfaces/IProxy.sol';
 import { IUpgradable } from '../interfaces/IUpgradable.sol';
 
+/**
+ * @title BaseProxy Contract
+ * @author Kiryl Yermakou
+ * @notice This abstract contract implements basic proxy functionalities. It provides an implementation
+ * address storage slot and a fallback function to delegate calls to the implementation address.
+ * This contract is meant to be inherited by other proxy contracts.
+ */
 abstract contract BaseProxy is IProxy {
     // bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)
     bytes32 internal constant _IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
     // keccak256('owner')
     bytes32 internal constant _OWNER_SLOT = 0x02016836a56b71f0d02689e69e326f4f4c1b9057164ef592671cf0d37c8040c0;
```

Comment:
set it to the keccak256 values instead of a hardcoded constant as a good pattern. this should be optimized during compilation

Reply:
When I change them back to keccak256 values I get this compilation error:
Only direct number constants and references to such constants are supported by inline assembly.